clients/java/src/main/java/io/camunda/zeebe/client/impl/command/CreateDocumentCommandImpl.java

```diff
@@ -155,33 +153,39 @@ public CreateDocumentCommandStep2 storeId(final String storeId) {
 
   @Override
   public CreateDocumentCommandStep2 contentType(final String contentType) {
-    metadata.put(DocumentMetadata.JSON_PROPERTY_CONTENT_TYPE, contentType);
+    metadata.setContentType(contentType);
     return this;
   }
 
   @Override
   public CreateDocumentCommandStep2 fileName(final String name) {
-    metadata.put(DocumentMetadata.JSON_PROPERTY_FILE_NAME, name);
+    metadata.setFileName(name);
     return this;
   }
 
   @Override
   public CreateDocumentCommandStep2 timeToLive(final Duration timeToLive) {
     final ZonedDateTime expiresAt = ZonedDateTime.now().plus(timeToLive);
-    metadata.put(DocumentMetadata.JSON_PROPERTY_EXPIRES_AT, expiresAt.toString());
+    metadata.setExpiresAt(expiresAt.toString());
     return this;
   }
 
   @Override
   public CreateDocumentCommandStep2 customMetadata(final String key, final Object value) {
     ensureNotNull("key", key);
-    metadata.put(key, value);
+    if (metadata.getCustomProperties() == null) {
+      metadata.setCustomProperties(new HashMap<>());
+    }
+    metadata.getCustomProperties().put(key, value);
     return this;
   }
 
   @Override
   public CreateDocumentCommandStep2 customMetadata(final Map<String, Object> customMetadata) {
-    metadata.putAll(customMetadata);
+    if (metadata.getCustomProperties() == null) {
```

Comment:
🔧 You can check if the provided `customMetadata` is not null, if not simply set it into `customProperties`

```
    ensuroNotNull("customeMetadata", customMetadata)
    metadata.setCustomProperties(customMetadata);
    return this;
```